src/main/resources/com/appian/intellij/k/k.flex

```diff
@@ -47,28 +47,51 @@ Q_SYSTEM_FUNCTION=(abs|acos|aj|aj0|all|and|any|asc|asin|asof|atan|attr|avg|avgs|
 
 Q_SQL_TEMPLATE=(select|exec|update|delete)
 
-INT_TYPE=[ihjepuvt]
-FLOAT_TYPE=[fen]
+INT_TYPE=[ihjfepnuvt]
+FLOAT_TYPE=[fe]
+TIMESPAN_STAMP_TYPE=[pn]
 Q_DATETIME_TYPE=[mdz]
-TYPE={INT_TYPE}|{FLOAT_TYPE}|{Q_DATETIME_TYPE}
+Q_TIME_TYPE=[uvt]
 // only positive numbers
-Q_NIL_POSITIVE="0N"({INT_TYPE}|{FLOAT_TYPE}|"g")
+Q_NIL=-?0[Nn]
 Q_BINARY=[01]"b"
 Q_HEX_CHAR=[[:digit:]A-Fa-f]
 Q_HEX_NUMBER="0x"{Q_HEX_CHAR}{Q_HEX_CHAR}?
-NUMBER_POSITIVE={Q_NIL_POSITIVE}|{Q_BINARY}|{Q_HEX_NUMBER}
+NUMBER_POSITIVE={Q_BINARY}|{Q_HEX_NUMBER}
 // positive & negative
-NIL_OR_INFINITY=0[iInNwW] // iI are from k3; wW are from k4; nN are from both
-Q_MONTH=[:digit:]+\.[:digit:]{2}
-Q_DATE={Q_MONTH}+\.[:digit:]{2}
+INFINITY=-?0[iIwW] // iI are from k3; wW are from k4; nN are from both
+Q_MONTH=[:digit:]{4}\.[:digit:]{2}
+Q_DATE={Q_MONTH}\.[:digit:]{2}
+Q_MINUTE=[:digit:]{2,3}\:[0-5][:digit:]
+Q_SECOND={Q_MINUTE}\:[0-5][:digit:]
 Q_TIME=[:digit:]+(\:[:digit:]{2}(\:[0-5][:digit:](\.[:digit:]+)?)?)?
 Q_DATETIME={Q_DATE}"T"{Q_TIME}
-NUMBER_POSITIVE_NEGATIVE=(0|[1-9][0-9]*)(\.[0-9]+)?([eE][+-]?[0-9]*)?|{NIL_OR_INFINITY}
+Q_TIMESTAMP=-?{Q_DATE}("D"{Q_TIME})?
+NUMBER_INTEGER=-?(0|[1-9][0-9]*)
+Q_TIMESPAN={NUMBER_INTEGER}"D"({Q_TIME})?
+NUMBER_FLOAT={NUMBER_INTEGER}(\.[0-9]*)|(\.[0-9]+)
+NUMBER_EXP={NUMBER_INTEGER}(\.[0-9]+)?([eE][+-]?[0-9]*)
 // all numbers
-NUMBER_NO_TYPE=-?({NUMBER_POSITIVE_NEGATIVE})|{NUMBER_POSITIVE}|-?({Q_DATE}|{Q_TIME}|{Q_DATETIME}|{Q_MONTH})
-NUMBER={NUMBER_NO_TYPE}{TYPE}?
+NUMBER={NUMBER_INTEGER}{INT_TYPE}?|({NUMBER_FLOAT}|{NUMBER_EXP}){FLOAT_TYPE}?|
+       ({INFINITY}|{Q_NIL})({INT_TYPE}|{Q_DATETIME_TYPE})?|{Q_NIL}"g"|
+       ({Q_TIMESTAMP}|{Q_TIMESPAN}|{NUMBER_FLOAT}){TIMESPAN_STAMP_TYPE}?|
+       {NUMBER_POSITIVE}|({Q_MINUTE}|{Q_SECOND}|{Q_TIME}){Q_TIME_TYPE}?|{Q_DATE}[dz]?|{Q_DATETIME}"z"?|{Q_MONTH}"m"?
 BINARY_VECTOR=[01][01]+"b"|"0x"{Q_HEX_CHAR}{2}{Q_HEX_CHAR}+
-NUMBER_VECTOR={NUMBER_NO_TYPE}({WHITE_SPACE}{NUMBER_NO_TYPE})+{TYPE}?|{BINARY_VECTOR}
+VECTOR_INT={NUMBER_INTEGER}|{Q_NIL}|{INFINITY}
```

Comment:
@kiniv (I thought I had added a comment here but I probably forgot to submit it), the naming change was around the rules of the form `VECTOR_{TYPE}`. Could we rename them to be `{TYPE}_VECTOR_ITEM` instead? e.g.
`VECTOR_INT` --> `INT_VECTOR_ITEM`